restricted_site_access.php

```diff
@@ -97,6 +97,7 @@ public static function add_actions() {
 
 		add_filter( 'pre_option_blog_public', array( __CLASS__, 'pre_option_blog_public' ), 10, 1 );
 		add_filter( 'pre_site_option_blog_public', array( __CLASS__, 'pre_option_blog_public' ), 10, 1 );
+		add_filter( 'redirect_canonical', array( __CLASS__, 'remove_redirect_guess_404_permalink' ) );
```

Comment:
`do_redirect_guess_404_permalink` is a more appropriate hook for this, it's been available since WordPress 5.5 so meets the plugin's minimum requirements.

Reply:
Didn't know this hook existed, thanks for suggesting this.